src/events.rs

```diff
@@ -71,7 +71,17 @@ impl EventEmitter {
     /// [`try_recv`]: Self::try_recv
     pub async fn recv(&self) -> Option<Event> {
         let mut lock = self.0.lock().await;
-        lock.recv().await.ok()
+        loop {
+            match lock.recv().await {
+                Err(async_broadcast::RecvError::Overflowed(_)) => {
+                    // Some events have been lost,
+                    // but the channel is not closed.
+                    continue;
```

Comment:
Maybe it's worth returning a special event telling that some events are lost? I.e. if we don't see such an event in the log, then no events are missing